spacy/language.py

```diff
@@ -265,7 +293,112 @@ def pipe_labels(self):
                 labels[name] = list(pipe.labels)
         return labels
 
-    def get_pipe(self, name):
+    @classmethod
+    def has_factory(cls, name: str) -> bool:
+        """RETURNS (bool): Whether a factory of that name is registered."""
+        return name in registry.factories
+
+    @classmethod
+    def get_factory_meta(cls, name: str) -> "FactoryMeta":
+        """RETURNS (FactoryMeta): The meta for the given factory name."""
+        if name not in cls._pipe_meta:
+            raise ValueError(Errors.E967.format(name=name))
+        return cls._pipe_meta[name]
+
+    @classmethod
+    def get_component_meta(cls, name: str) -> "FactoryMeta":
+        """RETURNS (FactoryMeta): The meta for the given component name."""
+        if name not in cls._component_meta:
+            raise ValueError(Errors.E967.format(name=name))
+        return cls._component_meta[name]
+
+    @classmethod
+    def factory(
+        cls,
+        name: str,
+        *,
+        default_config: Dict[str, Any] = SimpleFrozenDict(),
+        assigns: Iterable[str] = tuple(),
+        requires: Iterable[str] = tuple(),
+        retokenizes: bool = False,
+        func: Optional[Any] = None,
+    ) -> Callable:
+        """Register a new pipeline component factory. Can be used as a decorator
+        on a function or classmethod, or called as a function with the factory
+        provided as the func keyword argument.
+        """
+        if not isinstance(name, str):
+            raise ValueError(Errors.E963.format(decorator="factory"))
+        if not isinstance(default_config, dict):
+            err = Errors.E962.format(
+                style="default config", name=name, cfg_type=type(default_config)
+            )
+            raise ValueError(err)
+        if cls.has_factory(name):
+            raise ValueError(Errors.E004.format(name=name))
+
+        def add_factory(factory_func: Callable) -> Callable:
+            argspec = inspect.getfullargspec(factory_func).args
+            if "nlp" not in argspec or "name" not in argspec:
+                raise ValueError(Errors.E964.format(name=name))
+            # Officially register the factory so we can later call
+            # registry.make_from_config and refer to it in the config as
+            # @factory = "spacy.Language.xyz". We use the class name here so
```

Comment:
drop the "@" ?

Reply:
Shouldn't this be `@factories`?

```suggestion
            # @factories = "spacy.Language.xyz". We use the class name here so
```